### PR TITLE
Adds k'ois contaminated phoron (1)

### PR DESCRIPTION
**Pull request by Sniblet**
#15418

> Wow! My very first PR here and it's kinda big! I'm a bad coder and phoron is important so please review a lot!!
> 
> This splits reagent-form phoron into four types:
> toxin/phoron, which is now abstract and does not appear ingame;
> toxin/phoron/base, which does all the normal unobtanium things that phoron does and is what you get from grinding it/injecting with a vaurca hardsuit/eating filtered k'ois bars/etc;
> toxin/phoron/kois, which is only good as phoron for vaurca consumption (no chemical reactions except sedantian firestorm, doesn't work for sabotaging light tubes or whatever, etc), works as a secondary mycosis vector, and spawns in unclean k'ois snacks, k'ois mycosis procs, and the plant itself;
> toxin/phoron/kois/black, which spawns in black k'ois and does black k'ois things.
> 
> All of these will show as "Phoron" to most chemical machines, but if analyzed by a chemmaster, k'ois variants will have a bonus description sentence about biological traces. Don't drink phoron! You can never be sure that it's safe- wait.
> 
> This doesn't add gaseous k'ois phoron, partly because oh my god that's a lot of work and I'm GOING to break something trying that, partly because it'd need separate art that looks less energetic, and partly because people who're playing with k'ois phoron aren't really talking to atmos that much. If you spill phoron/kois into the air you get "safe" and functional phoron gas.
> Look.
> This is an improvement.
> I mean, if it works.
> 😩

### CODE REVIEW

**Comment by Sniblet** [comment]
!review

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm [hunk 1]
```diff
@@ -158,6 +158,52 @@
 	T.assume_gas(GAS_PHORON, amount, T20C)
 	remove_self(amount, holder)
 
+/singleton/reagent/toxin/phoron/kois
+	name = "Phoron"
+	description = "Phoron in its liquid form. Twice as potent when breathed in. Contains biological traces."
+	fallback_specific_heat = 1 //contaminated phoron acts more like k'ois than phoron
+	var/kois_type = 1
```
Comment (NonQueueingMatt) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:165]: this needs to be a define like KOIS_TYPE_NORMAL

File: code/modules/detectivework/tools/rag.dm [hunk 2]
```diff
@@ -228,10 +228,10 @@
 		return
 
 	//also copied from matches
-	if(REAGENT_VOLUME(reagents, /singleton/reagent/toxin/phoron)) // the phoron explodes when exposed to fire
+	if(REAGENT_VOLUME(reagents, /singleton/reagent/toxin/phoron/base)) // the phoron explodes when exposed to fire
```
Comment (NonQueueingMatt) [inline code/modules/detectivework/tools/rag.dm:231]: Surely this shouldn't be checking specifically for normal phoron? All phoron should be flammable/dangerous

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 3]
```diff
@@ -200,9 +200,9 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 		playsound(src, 'sound/items/cigs_lighters/cig_light.ogg', 75, 1, -1)
 		src.reagents.set_temperature(T0C + 45)
 		damtype = "fire"
-		if(REAGENT_VOLUME(reagents, /singleton/reagent/toxin/phoron)) // the phoron explodes when exposed to fire
+		if(REAGENT_VOLUME(reagents, /singleton/reagent/toxin/phoron/base)) // the phoron explodes when exposed to fire
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/cigs_lighters.dm:203]: Same as above. Also, "base" phoron is a pretty bad name - maybe "pure" would work better.

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

File: code/modules/detectivework/tools/rag.dm (hunk 2 above)
Comment (Sniblet) [reply code/modules/detectivework/tools/rag.dm:231]: If all phoron is flammable why is k'ois phoron useless as fuel

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm (hunk 1 above)
Comment (Sniblet) [reply code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:165]: var/kois_type = KOIS_TYPE_NORMAL?

File: code/game/objects/items/weapons/cigs_lighters.dm (hunk 3 above)
Comment (Sniblet) [reply code/game/objects/items/weapons/cigs_lighters.dm]: Same as above, and changed

**Review by Alberyk — APPROVED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**